### ISSUE TITLE
Resolve discrepancies between reported volumetric strains in SM and TM

### ISSUE THREAD

**Issue by bwspenc** [idaholab/moose#11743]
## Rationale

SolidMechanics and TensorMechanics use different procedures to apply volumetric eigenstrains. They appear to give the same response, but the quantity reported when evaluating the volumetric strain from the strain tensor differs.

## Description

Excerpt from an email from @jasondhales:
In SM, the component of the eigenstrain is ec_sm = (eV+1)^(1/3) - 1 (= L/L0-1 = (L-L0)/L0).

In TM, the component of the eigenstrain is ec_tm = log((eV+1)^(1/3)) = log(ec_sm+1) (= log(L/L0)).

So, from the beginning, the components of eigenstrains are different.  Since this is a stress-free test involving pure volumetric expansion, the strain fed to the constitutive law must be zero.  So, the total strain must equal the eigenstrain for both the SM and TM cases.  So, the total strain for SM and TM cannot be the same because ec_sm does not equal ec_tm.

The calculation of volumetric strain in the Calculator and Tools is based on the definition of volumetric strain being (V-V0)/V0.  This lines up with the SM calculation using an engineering strain definition.  The test in SM uses the linear option, and everything comes out perfectly.  If you change linear to nonlinear3d, the strains come out the same (they must since they must match the eigenstrain for this stress-free problem), but the volume is a little off (same values for strain, different definition for strain).  The reported volumetric strain still matches the analytical value since it is based on the strains themselves and not on the volume.  In other words, the reported volume does not match the reported volumetric strain based on the (V-V0)/V0 volumetric strain definition.

The TM version of this test uses FINITE as the strain option (logarithmic).  The computed strains match what is expected given the definition of ec_tm above.  The volume comes out correctly, as it should since the eigenstrain and total strain are logarithmic.  The reported volumetric strain does not match that in SM since the strains do not match those in SM.  If SMALL is used instead of FINITE, the strains do not change (they must match the eigenstrain, which doesn't change), but the volume is now off (same values for strain, different definition for strain).  Of course, the reported volumetric strain is different again.

What to do?

First, I haven't checked whether other eigenstrain models enforce a logarithmic strain, but maybe they should not.  This one is defined using (V-V0)/V, and an argument can be made that it shouldn't be adjusted.  I know we've been around this before.

Second, if we want a volumetric strain to come out of the code defined as (V-V0)/V0, we'll either have to base it on volumes or know what strain definition was used in computing the strain tensor and then unravel that if it isn't engineering strain.

## Impact

When this is sorted out, this will potentially change the reported volumetric strain from the material model in some cases in both solid_mechanics and tensor_mechanics.

**Comment by sapitts** [comment]
Test idea to try based on standup parking lot with @bwspenc and @acasagran: try running the same problem with the thermal strains but just with displacement boundary conditions to check the volume change.

**Comment by bwspenc** [comment]
To try to get to the bottom of this, I started looking at this test:
`moose/modules/tensor_mechanics/test/tests/volumetric_eigenstrain`

This is a test of the volumetric eigenstrain in TM.  It is a unit cube that has a very large volumetric eigenstrain applied to it using a function ramped up over two steps.  After two steps, the volumetric strain is 2.  The corresponding elongations of the edges is 0.44224957030741, which gives a final volume of 3: (1+ 0.44224957030741)^3 = 3. The test correctly computes the final volume as 3 using the VolumePostprocessor.  If you use the `VolumetricStrain` option in `RankTwoScalarAux`, in the absence of shear strains, it treats the strains (eps_xx = eps_yy = eps_zz = 0.366204) as linear strains, and incorrectly reports a volumetric strain of 1.55004. 

To understand whether there was something different about eigenstrains vs mechanical strains, I modified the same problem to remove the eigenstrain, and impose a prescribed displacement on the sides of the cube opposite to the fixed sides equal to the elongation computed by the original version of the model (0.44224957030741). I get exactly the same thing: VolumePostprocessor reports a final volume of 3 and `RankTwoScalarAux` with the `VolumetricStrain` option reports a volumetric strain of 1.55004.

Next, I converted the two models to SolidMechanics to understand what that system does differently.  For the eigenstrain variant of the test, SM doesn't offer a way to directly impose a volumetric strain. SM simply takes the linear strain due to thermal expansion and subtracts that from the total strain.  Since the total strain in SM (and TM) is logarithmic, I converted the linear expansion in the TM variant of the test to a logarithmic strain: log(1 + 0.44224957030741) = 0.3662040962227044 and applied that as the thermal expansion.  When I do that, I get exactly the same results as in the TM test: VolumePostprocessor reports a final volume of 3 and `MaterialTensorAux` with the `VolumetricStrain` option reports a volumetric strain of 1.55004.  I also got the same result with the mechanically imposed strain option in SM.

Bottom line: Both systems are behaving the same, and both systems are incorrectly treating the total strain tensor as if it were engineering strain, when it is really logarithmic strain, when they compute the volumetric strain for output. I think that is the thing that should be corrected.  @jasondhales, It's been so long since we first discussed this, that I don't remember what problem actually brought this up.  Do you have the test case that you used?

**Comment by jasondhales** [comment]
I'm late, obviously, but I think Ben's explanation is spot on. The strain tensor is logarithmic, and the calculation of volumetric strain assumes engineering.
